libpod/image/image.go

```diff
@@ -1006,63 +1068,164 @@ func splitString(input string) string {
 // IsParent goes through the layers in the store and checks if i.TopLayer is
 // the parent of any other layer in store. Double check that image with that
 // layer exists as well.
-func (i *Image) IsParent() (bool, error) {
-	children, err := i.GetChildren()
+func (i *Image) IsParent(ctx context.Context) (bool, error) {
+	children, err := i.getChildren(ctx, 1)
 	if err != nil {
 		return false, err
 	}
 	return len(children) > 0, nil
 }
 
+// historiesMatch returns the number of entries in the histories which have the
+// same contents
+func historiesMatch(a, b []imgspecv1.History) int {
```

Comment:
It would be nice to have this, historiesDiffer (from Buildah) in a common area, or perhaps c/image?  Not a todo for now, but maybe an intern task to move similar functions from here and buildah to there?